flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/SimpleConsumerThread.java

```diff
@@ -123,6 +124,7 @@ public SimpleConsumerThread(
 		this.fetchSize = getInt(config, "fetch.message.max.bytes", 1048576);
 		this.bufferSize = getInt(config, "socket.receive.buffer.bytes", 65536);
 		this.reconnectLimit = getInt(config, "flink.simple-consumer-reconnectLimit", 3);
+		this.clientId = config.getProperty("group.id", "flink-kafka-consumer-legacy-" + broker.id());
```

Comment:
If it is about client id in Kafka `SimpleConsumer`, why is this option named `group.id` and not `client.id`? As I understand, Flink job can be just one of consumers in a group.

Reply:
Hi, @azagrebin

group.id is a concept of kafka, used to describe consumers property, its official description is "A string that uniquely identifies the group of consumer processes to which this consumer belongs. By setting the same group id multiple processes indicate that they are all part of the same consumer group".

client.id is a concept of SimpleConsumer, its official descriptionis "The client id is a user-specified string sent in each request to help trace calls. It should logically identify the application making the request". its default value is "group.id value"

thanks.

Reply:
Hi @maqingxiang,

I would suggest to name it `client.id` as in its original description, because it is used in `SimpleConsumer` eventually in this case and `FlinkKafkaConsumer08` is based on the `SimpleConsumer`. As `client.id` identifies application, multiple `group.id`'s can have the same `client.id`. It should also help in logs and can be set to the same value as `group.id` in the user config if needed. 

`group.id` is also used in the commitment of the offsets in zookeeper for `FlinkKafkaConsumer08`. It is checked in `ZookeeperOffsetHandler` to be always set. So if we use it for the `clientId`, `"flink-kafka-consumer-legacy-" + broker.id()` will be ignored and `client.id` will be coupled with `group.id` with no way to use it for application name as original documentation suggests.

What do you think?

Cheers

Reply:
Why is it not
```
this.clientId = config.getProperty("client.id", "flink-kafka-consumer-legacy-" + broker.id());
```

that way the client ID would be configurable separately from the group ID. As it is in the PR, you can never have a client ID that is different from the group ID if you set a `group.id`.

Reply:
@maqingxiang 
Now I see your point, I think it makes sense to have `group.id` as a default value for `client.id` but keep them separately configurable, e.g.:
```
String groupId = config.getProperty("group.id", "flink-kafka-consumer-legacy-" + broker.id());
this.clientId = config.getProperty("client.id", groupId);
```

Reply:
Thank you for the review @azagrebin  and @aljoscha.

 I looked at the kafka document again, for `kafka consumer`, group.id is a required property. 

- for old consumer, the default value set "group.id value".
- for new consumer, the default value set "".

so, i agree with @aljoscha, clear and concise.
```
this.clientId = config.getProperty("client.id", "flink-kafka-consumer-legacy-" + broker.id());
```
If ok, I'll resubmit it again, thanks.

Reply:
I think it is better, but this PR change is in old consumer, I would make `group.id` default value for `client.id`, same as in Kafka docs. @aljoscha?

Reply:
sorry about all the confusion. I think what @azagrebin suggested is the way to go:
```
String groupId = config.getProperty("group.id", "flink-kafka-consumer-legacy-" + broker.id());
this.clientId = config.getProperty("client.id", groupId);
```

@maqingxiang Isn't this also what you initially suggested?

I'm afraid we just had a lot of miscommunication. 😅

Reply:
The way that @azagrebin is suggesting is really my initially suggested, but ` "flink-kafka-consumer-legacy-" + broker.id() ` has not been in effect.  

I'm a little confused now, how do I set it up better.

Reply:
I think we can just replace this line in the current PR:
```
this.clientId = config.getProperty("group.id", "flink-kafka-consumer-legacy-" + broker.id());
```
with these 2 lines:
```
String groupId = config.getProperty("group.id", "flink-kafka-consumer-legacy-" + broker.id());
this.clientId = config.getProperty("client.id", groupId);
```
`Group id` will be default value for `client id` then, as in Kafka docs, but user can change both of them independently if user ever needs it. It should also help in logs with multiple topics.
`"flink-kafka-consumer-legacy-" + broker.id()` will be just default value for `group.id`.